scripts/13-why-rust.md

```diff
@@ -0,0 +1,469 @@
+
+It has been the most loved language according to the largest developer survey on the planet for 7 years, that's every single year since 1.0.
+
+---
+
+![[jack-rust-perfect.png]]
+
+<https://twitter.com/jack/status/1474263588651126788>
+
+notes:
+
+Why does everyone love Rust?
+
+This is Jack Dorsey, who built twitter.
+
+---
+
+![[failwhale.png]]
+
+notes:
+
+Twitter started out using Ruby on Rails, then switched to a scala backend and microservices.
+
+Jack didn't build twitter with C or C++. He's a web developer.
+
+---
+
+> If you almost never add anything, you get C.
+> If you almost never remove anything, you get C++.
+
+&mdash;Raffi Molero
+
+notes:
+
+I don't quite agree that rust is a PERFECT language, there are many features that I want in Rust, and every 6 weeks I get them, thanks to the incredible core team.
+
+I don't think Rust is perfect, I think that Rust can allow you to write perfect code.
+
+---
+
+<split even>
+
+<ul>
+<li>Correctness focus</li>
+<li>Algebraic data types</li>
+<li>Compile-time execution</li>
+<li>Built-in transpiler metaprogramming</li>
+<li>Monads (shhh!)</li>
+</ul>
+
+<ul>
+<li>No runtime overhead</li>
+<li>No classes</li>
+<li>No runtime</li>
+<li>No bytecode</li>
+<li>No JIT</li>
+<li>No Garbage Collector</li>
+
+</ul>
+
+</split>
+
+notes:
+
+Rust's compromises are EXCELLENT.
+
+The language has all the features I want, and none of the ones I don't.
+
+Rust can basically do EVERYTHING.
+
+
+---
+
+> My biggest compliment to Rust is that it's boring, and this is an amazing compliment.
+
+– Chris Dickinson, on internal npm Rust adoption
+
+notes:
+
+It's not even that interesting.
+Rust isn't trendy, it's boring.
+
+---
+
+# Here Be Kraken
+
+notes:
+
+You know what isn't boring? seafaring in the ancient Mediterranean.
+
+If you were a sailor or at merchant, you would understand that you would lose ships to the sea now and then.
+
+So it was, so it had always been.
+
+Most of the sea was unexplored, and there were monsters, no doubt, taking the ships.
+
+Sailors accepted that these were just the risks that you had to accept to operate on the high seas.
+
+---
+
+# Here Be Overheads
+
+notes:
+
+You know what else isn't boring? SOFTWARE.
+
+And I don't mean in a good way.
+
+In our field in 2022, everyone accepts that there are some things you just have to deal with in order to write software.
+
+---
+
+# Lies Programmers Believe About Software
+
+notes:
+
+some lies programmers believe about software, include, but are not limited to:
+
+- That software is built on unknowable layers of abstraction
+- The layers can only do what you want if you recite the correct inscrutable incantations
+- Scaling horizontally is the way to solve your problems
+- Bugs must be caught in production.
+- If you don't use yaml and javascript to solve your problems, you will have to use xml and java.
+- speed
```

Comment:
I'm... not sure about those lies. I'm interested in what you've seen, because these seem a bit weird to me. Especially "bugs must be caught in production", that seems like something no software developer would believe no matter the experience...

Reply:
Agreed, I've changed it to "syntax is your enemy"